pkg/filesystem/local_directory_linux.go

```diff
@@ -2,5 +2,21 @@
 
 package filesystem
 
+import (
+	"os"
+	"runtime"
+
+	"golang.org/x/sys/unix"
+)
+
 // deviceNumber is the equivalent of POSIX dev_t.
 type deviceNumber = uint64
+
+func (d *localDirectory) Mknod(name string, perm os.FileMode, dev int) error {
+	if err := validateFilename(name); err != nil {
+		return err
+	}
+	defer runtime.KeepAlive(d)
+
+	return unix.Mknodat(d.fd, name, uint32(perm), dev)
```

Comment:
So the idea here is to use `os.FileMode`, which I think is perfect! The only thing that concerns me is that it may not necessarily be valid to convert `perm` to a uint32 directly. According to [the documentation](https://golang.org/pkg/os/#FileMode), bits like `ModeCharDevice` have values that are identical across platforms. Its value isn't necessarily the same as `unix.S_IFCHR`.

Should we add logic to switch on `perm&os.ModeType` and validate that that this is equal to `os.ModeDevice|os.ModeCharDevice`?